src/v/security/audit/schemas/utils.cc

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "security/audit/schemas/utils.h"
+
+namespace {
+
+namespace sa = security::audit;
+
+sa::api_activity::activity_id
+http_method_to_activity_id(std::string_view method) {
+    return string_switch<sa::api_activity::activity_id>(method)
+      .match_all("POST", "post", sa::api_activity::activity_id::create)
+      .match_all("GET", "get", sa::api_activity::activity_id::read)
+      .match_all("PUT", "put", sa::api_activity::activity_id::update)
+      .match_all("DELETE", "delete", sa::api_activity::activity_id::delete_id)
+      .default_match(sa::api_activity::activity_id::unknown);
+}
+
+sa::http_request from_ss_http_request(const ss::http::request& req) {
+    using ss_http_headers_t = decltype(req._headers);
+    const auto get_headers = [](ss_http_headers_t headers) {
+        std::vector<sa::http_header> audit_headers;
+        std::transform(
+          headers.begin(),
+          headers.end(),
+          std::back_inserter(audit_headers),
+          [](const auto& kv) {
+              return sa::http_header{.name = kv.first, .value = kv.second};
+          });
+        return audit_headers;
+    };
+    return sa::http_request{
+        .http_headers = get_headers(req._headers),
+        .http_method = req._method,
+        .url = {.hostname = "", .path = req.get_url(), .port = sa::port_t{req.get_client_address().port()}, .scheme = req.get_protocol_name(), .url_string = req.get_url()},
```

Comment:
clang-tidy may accept it, but let's fix it. (hint: i think you can add a comma after the last initializer)